ckanext/datavicmain/views/datavic_main.py

```diff
@@ -134,11 +134,32 @@ def admin_report():
         return response
     return render('admin/admin_report.html', extra_vars={})
 
+def nominate_view(package_id,view_id):
+    try:
+        toolkit.get_action('nominate_resource_view')({}, {'package_id': package_id, 'view_id':view_id})
+        toolkit.h.flash_success('Successfully nominated view: %s' % view_id)
+    except Exception as e:
+        print(str(e))
+        toolkit.h.flash_error('Exception')
+
+    return toolkit.h.redirect_to(f'/dataset/{package_id}')
+
+def denominate_view(package_id,view_id):
+    try:
+        toolkit.get_action('nominate_resource_view')({}, {'package_id': package_id, 'view_id':''})
+        toolkit.h.flash_success('Successfully denominated view: %s' % view_id)
+    except Exception as e:
+        print(str(e))
+        toolkit.h.flash_error('Exception')
+
+    return toolkit.h.redirect_to(f'/dataset/{package_id}')
 
 def register_datavicmain_plugin_rules(blueprint):
     blueprint.add_url_rule('/dataset/<id>/historical', view_func=historical)
     blueprint.add_url_rule('/dataset/purge/<id>', view_func=purge)
     blueprint.add_url_rule('/ckan-admin/admin-report', view_func=admin_report)
+    blueprint.add_url_rule('/dataset/<package_id>/nominate_view/<view_id>', view_func=nominate_view)
```

Comment:
Looks like your view can change the state of the dataset even if a user visits the page via GET method. It's not secure

Reply:
`POST`.